lf-edge/eve
# Add a host<->vm communication channel (vsock)

The vcomlink agent is responsible for listening on a vsock channel and server requests from VMs. The initial implementation only supports a TPM command to send the host's TPM endorsement key to the VM.

This is part of some primarily work for https://github.com/lf-edge/eve/pull/4071

## TODO
- [x] ~using protobuf~ reverted back to json due to unnecessary complexity and hacks.
- [x] set read/write timeout for socket
- [x] remove the wait on GCI
- [x] ~encoded packet size in as a header~ reverted back to json due to unnecessary complexity
- [x] add a doc commit after the code review is done
- [x] ~add client code~ will add in another PR (or repo)

## Eden Test
https://github.com/lf-edge/eden/pull/1016

## Manual Test

1- the host is listening on the vsock CID/Port as expected :
```
7376deaf-a849-4336-8112-6dfcba444e86:~# ss -l --vsock
Netid  State   Recv-Q  Send-Q   Local Address:Port   Peer Address:Port Process
v_str  LISTEN  0       0                    2:2000              *:*
```

2- server is reachabe from a VM, here I sent a `TpmRequestGetEk` to `ChannelTpm` : 
```
7376deaf-a849-4336-8112-6dfcba444e86:~# eve attach-app-console 7da404db-e8bb-45f6-bbf5-696768927e17.1.1/cons
[12:13:44.460] tio v1.37
[12:13:44.460] Press ctrl-t q to quit
[12:13:44.460] Connected

ubuntu@vm01:~$ ubuntu
ubuntu: command not found
ubuntu@vm01:~$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.4 LTS
Release:	22.04
Codename:	jammy
ubuntu@vm01:~$ echo '{"channel":2,"request":1}' | socat - VSOCK-CONNECT:2:2000
{"ek":"AAEACwADALIAIINxl2dEhLP4GpDMjUal1yT9UtduBlILZPKh2hszFGmqAAYAgABDABAIAAAAAAABAN+0/EsS5r8dp70gZvV6df5Dkq//Jnk4/h52ELp2+VxVdw7iTXe/R+Tcr2dHc5NB/eIoe5st06cgyB+L/lBoXH44xvx/FqWx1RkGBbb97C5kHOFgnK7OjvJmbEST1iVANYMzfk7L8AdR+YkxQ/jiABTBqlpWhZ/pxRuUOQddZkduXFTrQux9+ibjelUBV0exwIKVmcjqh3+n8LGZkE6uL2Dev1av82ykPAc+S2DCKSibHx2V+tG44ROVxywgZlvGxMJi8eU5KKAs/NAzhfeSui/hymv3CUDppTNI8oXZIZxk4WlzDA1W1MGbZLWv1z3gI2DiacqjuPuHjV5eNALk2fc="}
ubuntu@vm01:~$ 
```
